modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1463,14 +1466,125 @@ export class Wallet implements IWallet {
+   * @param {string} [params.shareOptions[].keychain.toPubKey] - The recipient's public key.
+   * @param {string} [params.shareOptions[].keychain.path] - The derivation path of the keychain.
+   * @param {string} params.shareOptions[].user - The user to share the wallet with.
+   * @param {string} params.shareOptions[].permissions - The permissions granted to the user.
+   * @param {string} [params.walletPassphrase] - The wallet passphrase used to decrypt the keychain.
+   * @throws {Error} If `shareOptions` is empty, or if required keychain parameters (`toPubKey` and `path`) are missing when needed.
+   * @throws {Error} If unable to decrypt the user keychain.
+   * @returns {Promise<CreateBulkWalletShareListResponse>} A promise that resolves with the response of the bulk wallet share creation.
+   */
+  async createBulkWalletShare(params: BulkWalletShareOptions): Promise<CreateBulkWalletShareListResponse> {
+    if (!params.keyShareOptions || Object.keys(params.keyShareOptions).length === 0) {
+      throw new Error('shareOptions cannot be empty');
+    }
+    const bulkCreateShareOptions: BulkCreateShareOption[] = [];
+
+    for (const shareOption of params.keyShareOptions) {
+      common.validateParams(shareOption, ['userId', 'pubKey', 'path'], []);
+
+      const needsKeychain = shareOption.permissions && shareOption.permissions.includes('spend');
+      let sharedKeychain;
+
+      if (needsKeychain) {
+        try {
+          const keychain = await this.getEncryptedUserKeychain();
+
+          if (keychain.encryptedPrv) {
+            const userPrv = decryptKeychainPrivateKey(this.bitgo, keychain, params.walletPassphrase);
+            if (!userPrv) {
+              throw new Error('Unable to decrypt user keychain.');
+            }
+
+            const ecdhkey = makeRandomKey();
+            assert(shareOption.pubKey, 'pubKey must be defined for sharing');
+
+            const secret = getSharedSecret(ecdhkey, Buffer.from(shareOption.pubKey, 'hex')).toString('hex');
+            const newEncryptedPrv = this.bitgo.encrypt({ password: secret, input: userPrv });
+
+            let pub = keychain.pub ?? keychain.commonPub;
+            if (keychain.commonKeychain) {
+              pub =
+                this.baseCoin.getMPCAlgorithm() === 'eddsa'
+                  ? EddsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain)
+                  : EcdsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain);
+            }
+
+            sharedKeychain = {
+              pub,
+              encryptedPrv: newEncryptedPrv,
+              fromPubKey: ecdhkey.publicKey.toString('hex'),
+              toPubKey: shareOption.pubKey,
+              path: shareOption.path,
+            };
+          }
+        } catch (e) {
+          if (e.message === 'No encrypted keychains on this wallet.') {
+            sharedKeychain = {};
+            // ignore this error because this looks like a cold wallet
+          } else {
```

Comment:
Isn't this the same code as this? https://github.com/BitGo/BitGoJS/blob/master/modules/sdk-core/src/bitgo/wallet/wallet.ts#L1507

Let's not copy paste. Preferably move it to a helper and re-use